### PR TITLE
TST: Add python 3.7 testing

### PR DESCRIPTION
**Pull request by thequackdaddy**
I think this will get python 3.7 testing on patsy

### CODE REVIEW

**Comment by thequackdaddy** [comment]
I also added a `try ... except` around the `import collections.abc from Mapping` as that got moved.

File: tools/check-API-refs.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 root = dirname(dirname(abspath(__file__)))
 patsy_ref = root + "/doc/API-reference.rst"
 
-doc_re = re.compile("^\.\. (.*):: ([^\(]*)")
```
Comment (njsmith) [inline tools/check-API-refs.py]: Better to make it a raw string, `r"..."`

File: .travis.yml [hunk 2]
```diff
@@ -23,6 +23,13 @@ matrix:
       env: PANDAS_VERSION_STR="NONE"
     - python: 2.7
       env: PANDAS_VERSION_STR="NONE"
+    - python: 3.7
+      sudo: true
+      dist: xenial
+    - python: 3.7
+      sudo: true
+      dist: xenial
+      env: PANDAS_VERSION_STR="NONE"
```
Comment (njsmith) [inline .travis.yml]: This is the simplest change that will work, but the overall situation is kind of silly. The only reason `sudo: true, dist: xenial` is needed is to get travis's version of 3.7, but we don't use travis's version of 3.7 for anything at all. We request it, force Travis to install it, and then we immediately throw it away and switch to conda. (And this is true for all the Python versions, just it becomes more obviously silly at the point where we start adding workarounds for limitations of Travis's Python...)

We'd be better off using `language: none`, and setting `PYTHON_VERSION` as an env-var... that's a larger change though.

File: tools/check-API-refs.py (hunk 1 above)
Comment (thequackdaddy) [reply tools/check-API-refs.py]: Changed.

File: .travis.yml (hunk 2 above)
Comment (thequackdaddy) [reply .travis.yml]: Agreed. I went ahead and pushed how I think you could do it without the hack. Honestly, its not much work and not that much of a larger change, so i'd suggest using it. You'll likely want to squash all my commits.

**Comment by njsmith** [comment]
Apparently I misremembered the syntax, and `language: none` actually means "ruby". The actual no-language images are `minimal` and `generic`: https://docs.travis-ci.com/user/languages/minimal-and-generic/

I just tried switching it to `minimal`, let's see what Travis does...

**Comment by Juanlu001** [comment]
Any chance to release these changes?

**Pull request merged**